src/providers/vmware/amd64.rs

```diff
@@ -39,8 +63,155 @@ impl VmwareProvider {
     fn fetch_guestinfo(erpc: &mut vmw_backdoor::EnhancedChan, key: &str) -> Result<Option<String>> {
         let guestinfo = erpc
             .get_guestinfo(key.as_bytes())
-            .context("failed to retrieve network kargs")?
+            .with_context(|| format!("failed to retrieve guestinfo for {}", key))?
             .map(|bytes| String::from_utf8_lossy(&bytes).into_owned());
         Ok(guestinfo)
     }
+
+    pub fn parse_netplan_config(&self) -> Result<Option<String>> {
+        if let Some(metadata) = &self.guestinfo_metadata {
+            // We need to parse the netplan config to remove the non-Netplan keys.
+            // The data can either be JSON or YAML, but since JSON is a subset of
+            // YAML we don't need to try serde_json::from_str here.
+            let netplan_config_unfiltered: serde_json::Value = serde_yaml::from_str(metadata)?;
+            // Only the "network" key is allowed to be present.
+            // We use the json! macro but this is only about creating a serde value::Value,
+            // even though its name sounds like it would create JSON.
+            let netplan_config_filtered = json!({ "network": netplan_config_unfiltered.get("network").ok_or(anyhow!("no 'network' key found"))?});
```

Comment:
For clarity, can we reformat this as

```rust
let ... = json!({
	...
})
```

?

I don't _think_ `cargo fmt` will revert it.

Reply:
Yes, works